server/mediaserver.go

```diff
@@ -368,7 +369,11 @@ func (s *LivepeerServer) registerConnection(rtmpStrm stream.RTMPVideoStream) (*r
 		pl:      core.NewBasicPlaylistManager(mid, storage),
 		profile: &vProfile,
 		lock:    &sync.RWMutex{},
-
+		sessManager: &BroadcastSessionsManager{
+			broadcastSessions: []*BroadcastSession{},
+			sessLock:          &sync.Mutex{},
+			broadcastSessMap:  make(map[string]*BroadcastSession),
```

Comment:
I think we can refine how we do this. In particular, let's:

* Clean  up initialization of the BroadcastSessionManager struct
* Eliminate passing around the LivepeerNode and the playlist, which is only there to propagate down to `selectOrchs` which makes it a bit smelly.
* Double down on the concept of abstracting away the `selectOrchs` function for testability / reusability
* Maybe eliminate the need for the code around startSession / startSessionListener.

```golang
type BroadcastSessionsManager  struct {
  sessions []*BroadcastSession{}
  sessionMap map[string]*BroadcastSession
  lock sync.Mutex
  createSessions func() []*BroadcastSession{}
}

func NewSessionManager(n *core.LivepeerNode, pl *core.PlaylistManager) *BroadcastSessionManager {
  m := &BroadcastSessionManager {
    // we really don't have to explicitly define additional fields here; zero values are fine
    // given the field definitions in BroadcastSessionManager above
    sessionMap: make(map[string]*BroadcastSession),
    createSessions: func() []*BroadcastSessionManager {
      return selectOrchestrator(node, pl)
    },
  }
  m.refreshSessions()
  return m
}

...

cxn := &rtmpConnection {
  ...
  sessManager = NewSessionManager(node, pl)
}
```

Reply:
This is very helpful feedback Josh! Always making me a better engineer